src/script/cpp_api/s_entity.cpp

```diff
@@ -198,14 +226,70 @@ void ScriptApiEntity::luaentity_Step(u16 id, float dtime,
 	luaL_checktype(L, -1, LUA_TFUNCTION);
 	lua_pushvalue(L, object); // self
 	lua_pushnumber(L, dtime); // dtime
-	/* moveresult */
-	if (moveresult)
-		push_collision_move_result(L, *moveresult);
-	else
+
+	pushFloatPos(L, pos); // pos
+	push_v3f(L, rotation * core::DEGTORAD); // rotation (in radians)
+	pushFloatPos(L, new_velocity); // new_velocity
+	pushFloatPos(L, old_velocity); // old_velocity
+
+	if (moveresult) {
```

Comment:
The `old_velocity` and `new_velocity` are entirely different from the results that this PR provides.  The distinction is even important since the `moveresult` is only available when the object is `physical=true`. Yet non-physical objects can have a change in velocity. So that is why I am reporting velocity changes separate from the move result.

In addition, the 'old_velocity' is a very misleading name since it's not actually the old velocity, it's the intended velocity [once acceleration has been applied](https://github.com/minetest/minetest/blob/80d9e9c1834fdc78588ffe2842766c0177227786/src/collision.cpp#L252) AND as a result of [each prior collision box processed](https://github.com/minetest/minetest/blob/80d9e9c1834fdc78588ffe2842766c0177227786/src/collision.cpp#L537) within the loop. As a result it's not dependent on any specific node/object collision, but rather on an implementation-specific order of collision processing which could change at any time in the future, making it highly volatile for use in modding.

As for`new_velocity` that is even less intuitive. Just like `old_velocity`, the value is entirely dependent on the sequence of collision boxes already processed within the loop but also  the current node/object collision, so it doesn't reflect anything about the object's "new velocity" that is particularly useful for modders.

I feel very strongly that modders should not care about either of these values. And relying on them is bound to lead to more frustration since nobody (outside of those familiar with the CPP internals) will comprehend these implementation details. I think it's far better to keep this API as straightforward as possible by exposing `new_vel` and `old_vel` which should cover the vast majority of use-cases that I've found over the course of the past year.

PS. Don't get me wrong. I appreciate the effort you invested in your earlier PR. I'm just of the opinion that the `move result` table could be better tuned for modding purposes rather than exposed as-is.